platforms/software/dependency-management/src/main/java/org/gradle/internal/component/external/model/ExternalComponentResolveMetadata.java

```diff
@@ -16,10 +16,77 @@
 
 package org.gradle.internal.component.external.model;
 
-import org.gradle.internal.component.model.ComponentResolveMetadata;
+import com.google.common.collect.ImmutableList;
+import org.gradle.api.artifacts.ModuleVersionIdentifier;
+import org.gradle.api.artifacts.component.ComponentIdentifier;
+import org.gradle.api.attributes.HasAttributes;
+import org.gradle.api.internal.attributes.AttributesSchemaInternal;
+import org.gradle.api.internal.attributes.ImmutableAttributes;
+import org.gradle.internal.component.model.ComponentArtifactResolveMetadata;
+import org.gradle.internal.component.model.ComponentArtifactResolveState;
+import org.gradle.internal.component.model.ComponentGraphResolveMetadata;
+import org.gradle.internal.component.model.ComponentGraphResolveState;
 import org.gradle.internal.component.model.ModuleSources;
 
-public interface ExternalComponentResolveMetadata extends ComponentResolveMetadata {
-    @Override
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * The meta-data for an external component instance that is required during dependency resolution.
+ *
+ * <p>Note that this type is being replaced by several other interfaces that separate out the data and state required at various stages of dependency resolution.
+ * You should try to use those interfaces instead of using this interface or introduce a new interface that provides a view over this type but exposes only the
+ * data required.
+ * </p>
+ *
+ * @see ComponentGraphResolveState
+ * @see ComponentGraphResolveMetadata
+ * @see ComponentArtifactResolveState
+ * @see ComponentArtifactResolveMetadata
+ */
+public interface ExternalComponentResolveMetadata extends HasAttributes {
+    List<String> DEFAULT_STATUS_SCHEME = Arrays.asList("integration", "milestone", "release");
+
+    /**
+     * Returns the identifier for this component.
+     */
+    ComponentIdentifier getId();
+
+    /**
+     * Returns the module version identifier for this component. Currently, this reflects the (group, module, version) that was used to request this component.
+     *
+     * <p>This is a legacy identifier and is here while we transition the meta-data away from ivy-like
+     * module versions to the more general component instances. Currently, the module version and component identifiers are used interchangeably. However, over
+     * time more things will use the component identifier. At some point, the module version identifier will become optional for a component.
+     */
+    ModuleVersionIdentifier getModuleVersionId();
+
+    /**
+     * @return the sources information for this component.
+     */
     ModuleSources getSources();
+
+    /**
+     * Returns the schema used by this component.
+     */
+    AttributesSchemaInternal getAttributesSchema();
+
+    /**
+     * Returns true when this metadata represents the default metadata provided for components with missing metadata files.
+     */
+    boolean isMissing();
```

Comment:
💭 Would `isDefault` be a better name?

Reply:
I am not completely sure how this is used and haven't dug into its purpose fully, so I'd prefer not to change it. 

Part of the complexity of this interface hierarchy is that multiple interfaces define the same method. This method is also defined in `ModuleComponentGraphResolveMetadata`, and the classes that implement `ExternalComponentResolveMetadata` also implement that interface too.

In any case, all the methods in `ExternalComponentResolveMetadata` were just copied from `ComponentResolveMetadata`. I want to avoid any supplemental changes here if possible

Reply:
Sounds reasonable.